resources/PythonPip3Dsc/PythonPip3Dsc.psm1

```diff
@@ -399,6 +485,28 @@ class Pip3Package
         }
     }
 
+    [string] WhatIf()
+    {  
+        if ($this.Exist)
+        {
+            $whatIfState = Invoke-Pip3Install -PackageName $this.PackageName -Version $this.Version -Arguments $this.Arguments -DryRun
+
+            $out = @{
+                PackageName = $this.PackageName
+                _metaData   = @{
+                    whatIf = $whatIfState.StdOut
+                }
+            }
+        }
+        else 
+        {
+            # Uninstall does not have --dry-run param
+            $out = @{}
+        }
+        
+        return ($out | ConvertTo-Json -Depth 10 -Compress)
```

Comment:
Do you have an example of this output and what it might look like? Could you also add a test that attempts to install an invalid package using WhatIf?

Reply:
I have added additional tests that capture the output now :)